contracts/FlowStakingCollection.cdc

```diff
@@ -0,0 +1,844 @@
+        /// Accounts without a locked account will not store this, it will be nil
+        access(self) var tokenHolder: Capability<&LockedTokens.TokenHolder>?
+
+        /// Tracks how many locked and unlocked tokens the staker is using for all their nodes and/or delegators
+        /// When committing new tokens, locked tokens are used first, followed by unlocked tokens
+        /// When withdrawing tokens, unlocked tokens are withdrawn first, followed by locked tokens
+        pub var lockedTokensUsed: UFix64
+        pub var unlockedTokensUsed: UFix64
+
+        init(unlockedVault: Capability<&FlowToken.Vault>, tokenHolder: Capability<&LockedTokens.TokenHolder>?) {
+            pre {
+                unlockedVault.check(): "Invalid FlowToken.Vault capability"
+            }
+            self.unlockedVault = unlockedVault
+
+            self.nodeStakers <- {}
+            self.nodeDelegators <- {}
+
+            self.lockedTokensUsed = 0.0
+            self.unlockedTokensUsed = 0.0
+
+            // If the account has a locked account, initialize its token holder
+            // and locked vault capability
+            if let tokenHolderObj = tokenHolder {
+                self.tokenHolder = tokenHolder
+
+                // borrow the main token manager object from the locked account 
+                // to get access to the locked vault capability
+                let lockedTokenManager = tokenHolderObj.borrow()!.borrowTokenManager()
+                self.lockedVault = lockedTokenManager.vault
+            } else {
+                self.tokenHolder = nil
+                self.lockedVault = nil
+            }
+        }
+
+        /// Close all the stakes before destroying everything
+        /// This uses the closeStake method, so it will panic if there are still tokens staked in any of the objects
+        destroy() {
+            let nodeIDs = self.getNodeIDs()
+            let delegatorIDs = self.getDelegatorIDs()
+
+            for nodeID in nodeIDs {
+                self.closeStake(nodeID: nodeID, delegatorID: nil)
+            }
+
+            for delegatorID in delegatorIDs {
+                self.closeStake(nodeID: delegatorID.delegatorNodeID, delegatorID: delegatorID.delegatorID)
+            }
+
+            destroy self.nodeStakers
+            destroy self.nodeDelegators
+        }
+
+        /// Called when committing tokens for staking. Gets tokens from either or both vaults
+        /// Uses locked tokens first, then unlocked if any more are still needed
+        access(self) fun getTokens(amount: UFix64): @FungibleToken.Vault {
+
+            // If there is a locked account, use the locked vault first
+            if self.lockedVault != nil {
+
+                var lockedBalance: UFix64 = self.lockedVault!.borrow()!.balance - FlowStorageFees.minimumStorageReservation
+                var unlockedBalance: UFix64 = self.unlockedVault.borrow()!.balance - FlowStorageFees.minimumStorageReservation
```

Comment:
Consider declaring a variable for the repeated `self.lockedVault!.borrow()!` and `self.unlockedVault.borrow()!` here and below (for the withdrawal).

Also, the variable declarations neither need to be mutable nor need a type annotation:

```suggestion
				let lockedVault = self.lockedVault!.borrow()!
				let unlockedVault = self.unlockedVault.borrow()!

                let lockedBalance = lockedVault.balance - FlowStorageFees.minimumStorageReservation
                let unlockedBalance = unlockedVault.balance - FlowStorageFees.minimumStorageReservation
```